### PR TITLE
Improved `needreport` directive by adding needs metrics analysis.

### PR DESCRIPTION
**Pull request by iSOLveIT**
Fix for issue #631.

### CODE REVIEW

**Review by danwos — APPROVED** [review]
Approved, but I have one question, and we could extend the available metrics.
But this can be done by  a new PR.

**Comment by iSOLveIT** [comment]
Yes. The other metrics can be included in a new PR later. This PR was intended to show the first implementation of the idea.

File: docs/directives/needreport.rst [hunk 1]
```diff
@@ -11,6 +11,13 @@ needreport
 * :ref:`Links <needs_extra_links>`
 * :ref:`Options <needs_extra_options>`
 
+and it also adds some needs metrics using the `usage`_ option.
+
+To use the ``needreport`` directive, you need to set the :ref:`needs_report_template`
```
[the remark this answers is not in the document]
Comment (iSOLveIT) [reply docs/directives/needreport.rst:16]: It is not mandatory. There is a default template to use if you don't set it.

[the remark this answers is not in the document]
Comment (danwos) [reply docs/directives/needreport.rst:16]: But it read like this :) (at least for me)

So you can use ``needreport`` without defining a template. That should be more clearly written.

[the remark this answers is not in the document]
Comment (iSOLveIT) [reply docs/directives/needreport.rst:16]: Yeah I have corrected that in one of the PRs.

**Pull request merged**